### PR TITLE
MacOS Comp Tests

### PR DESCRIPTION
**Pull request by aburrell**
Added compilation tests for the most recent MacOS system.  This should help us catch updates that affect users when they occur.

### CODE REVIEW

**Comment by aburrell** [comment]
All tests will pass after #622 is merged into this branch.  Alternatively, close #622 and just merge this into develop.

**Comment by pasha-ponomarenko** [comment]
@aburrell , unfortunately I cannot test it as do not use Mac... :-(

**Comment by mtwalach** [comment]
Hi @aburrell - I tried testing this but came across a few issues:

1) problem with /codebase/base/src.lib/graphic/xwin.1.5/: 
This was expecting $(XPATH)/include/include/X11...

So I changed `/rst-macos_comp_tests/codebase/base/src.lib/graphic/xwin.1.5/src/makefile`

From 

`INCLUDE=-I$(IPATH)/base -I$(XPATH)/include`

To 

`INCLUDE=-I$(IPATH)/base -I$(XPATH)`


2) problem with codebase/analysis/src.lib/cdf/rcdf.1.5/:

Same thing with the CDF_PATH, so I changed `/rst-macos_comp_tests/codebase/analysis/src.lib/cdf/rcdf.1.5/src/makefile`

From 

`INCLUDE=-I$(IPATH)/analysis -I$(CDF_PATH)/include`

To 

`INCLUDE=-I$(IPATH)/analysis -I$(CDF_PATH)`


3) Another hiccup:
================================================================================
Compiling Binary:/Users/walach/Documents/rst-macos_comp_tests/codebase/base/src.bin/graphic/pngd.1.2
================================================================================
```
make clean
rm -f *.o
rm -f version.h
rm -f errstr.h
rm -f hlpstr.h
rm -f pngd
make
make.version /Users/walach/Documents/rst-macos_comp_tests/codebase/base/src.bin/graphic/pngd.1.2
make.help
./doc/pngd.doc.xml
cc -Wall -pedantic -O3 -D_GNU_SOURCE -D_DARWIN -I/opt/X11/include/include -D_XLIB_ -D_SVGLIB_ -I/Users/walach/Documents/rst-macos_comp_tests/include/base -c pngd.c    
pngd.c:30:10: fatal error: 'X11/Xlib.h' file not found
   30 | #include <X11/Xlib.h>
      |          ^~~~~~~~~~~~
1 error generated.
make: *** [pngd.o] Error 1
Compilation Aborted.
```

I couldn't figure out how to remove the extra "include" here so I made a recursive symlink: 

sudo ln -s /opt/X11/include /usr/X11R6
sudo ln -s /usr/X11R6/include /usr/X11R6/include


4) Similar thing:
================================================================================
Compiling Binary:/Users/walach/Documents/rst-macos_comp_tests/codebase/general/src.bin/dmap/dmaptocdf.1.9
================================================================================
```
make clean
rm -f *.o
rm -f version.h
rm -f errstr.h
rm -f hlpstr.h
rm -f dmaptocdf
make
make.version /Users/walach/Documents/rst-macos_comp_tests/codebase/general/src.bin/dmap/dmaptocdf.1.9
make.help
./doc/dmaptocdf.doc.xml
cc -Wall -pedantic -O3 -D_GNU_SOURCE -D_DARWIN -I/opt/X11/include/include -D_SVGLIB_ -I/Users/walach/Documents/rst-macos_comp_tests/include/base -I/Users/walach/Documents/rst-macos_comp_tests/include/general -I/Applications/cdf/cdf39_0-dist/include/include -c dmaptocdf.c    
dmaptocdf.c:34:10: fatal error: 'cdf.h' file not found
   34 | #include "cdf.h"
      |          ^~~~~~~
1 error generated.
make: *** [dmaptocdf.o] Error 1
Compilation Aborted.
```


Fixed with
`sudo ln -s /Applications/cdf/cdf39_0-dist/include /Applications/cdf/cdf39_0-dist/include`

5) Here I am stuck: 
================================================================================
Compiling Binary:/Users/walach/Documents/rst-macos_comp_tests/codebase/general/src.bin/dmap/dmaptoncdf.1.11
================================================================================

```
make clean
rm -f *.o
rm -f version.h
rm -f errstr.h
rm -f hlpstr.h
rm -f dmaptoncdf
make
make.version /Users/walach/Documents/rst-macos_comp_tests/codebase/general/src.bin/dmap/dmaptoncdf.1.11
make.help
./doc/dmaptoncdf.doc.xml
cc -Wall -pedantic -O3 -D_GNU_SOURCE -D_DARWIN -I/opt/X11/include/include -D_SVGLIB_ -I/Users/walach/Documents/rst-macos_comp_tests/include/base -I/Users/walach/Documents/rst-macos_comp_tests/include/general -I/usr/local/include/include -c dmaptoncdf.c    
In file included from dmaptoncdf.c:34:
/usr/local/include/netcdf.h:1946:24: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
 1946 | EXTERNL int nc_finalize();
      |                        ^
      |                         void
In file included from dmaptoncdf.c:37:
/Users/walach/Documents/rst-macos_comp_tests/include/general/dmap.h:104:28: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
  104 | struct DataMap *DataMapMake();
      |                            ^
      |                             void
2 warnings generated.
mkdir -p /Users/walach/Documents/rst-macos_comp_tests/bin
cc -Wl,-rpath,@loader_path/../lib/ -Wl,-rpath,@loader_path/../../lib/ -L/opt/X11/include/lib -L/Users/walach/Documents/rst-macos_comp_tests/lib -o /Users/walach/Documents/rst-macos_comp_tests/bin/dmaptoncdf dmaptoncdf.o -lopt.1 -ldmap.1 -lrcnv.1  \
                            -lm -lz -L/usr/local/include/lib -lnetcdf 
ld: warning: search path '/usr/local/include/lib' not found
ld: warning: ignoring file '/usr/local/Cellar/netcdf/4.5.0/lib/libnetcdf.13.dylib': found architecture 'x86_64', required architecture 'arm64'
Undefined symbols for architecture arm64:
  "_nc_close", referenced from:
      _main in dmaptoncdf.o
  "_nc_inq_dimlen", referenced from:
      _main in dmaptoncdf.o
  "_nc_inq_vardimid", referenced from:
      _main in dmaptoncdf.o
  "_nc_inq_varid", referenced from:
      _main in dmaptoncdf.o
      _main in dmaptoncdf.o
  "_nc_inq_varndims", referenced from:
      _main in dmaptoncdf.o
  "_nc_open", referenced from:
      _main in dmaptoncdf.o
  "_nc_put_var1_double", referenced from:
      _main in dmaptoncdf.o
  "_nc_put_var1_float", referenced from:
      _main in dmaptoncdf.o
  "_nc_put_var1_int", referenced from:
      _main in dmaptoncdf.o
  "_nc_put_var1_short", referenced from:
      _main in dmaptoncdf.o
  "_nc_put_var1_text", referenced from:
      _main in dmaptoncdf.o
  "_nc_put_vara_double", referenced from:
      _main in dmaptoncdf.o
  "_nc_put_vara_float", referenced from:
      _main in dmaptoncdf.o
  "_nc_put_vara_int", referenced from:
      _main in dmaptoncdf.o
  "_nc_put_vara_short", referenced from:
      _main in dmaptoncdf.o
  "_nc_put_vara_text", referenced from:
      _main in dmaptoncdf.o
      _main in dmaptoncdf.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [dmaptoncdf] Error 1
Compilation Aborted.
```

Is this a problem with my netcdf installation?

**Comment by aburrell** [comment]
Hi @pasha-ponomarenko and @mtwalach this pull request allows compilation testing for Mac OS on the GitHub Actions system.  It won't necessarily work on an individual's system because of the path hiccups @mtwalach identified.  If you look at the checks above, the [pull request mac test](https://github.com/SuperDARN/rst/actions/runs/11153340867/job/31000648263?pr=621) shows these changes succeed, while the [push mac test](https://github.com/SuperDARN/rst/actions/runs/11153340655/job/31000647724?pr=621) shows that the tests cannot pass without the profile parameters provided in the `.github/test_profiles` directory.

**Comment by aburrell** [comment]
@mtwalach, basically, you do what you normally do to install things.  This won't fully work unless #622 is merged into this branch first (since it has the fixes that were made to the complex numbers that caused problems with the recent mac operating systems).  The changes here will not impact the user installation, it will only enable the GitHub Action tests to run.  That's why all the files added/changes are hidden files.  Hit me up on slack/email if you want to know more on how this works!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -32,15 +32,41 @@ jobs:
...
+      if: startsWith(matrix.os, 'mac')
       run: |
         export RSTPATH=$(pwd)
+        cp .github/test_profiles/profile.bash.darwin .profile.bash
+        cp .github/test_profiles/base.bash.darwin .profile/base.bash
+        cp .github/test_profiles/makelib.darwin build/make/makelib.darwin
         source .profile.bash
+        echo $SYSTEM
+        echo $OSTYPE
+        echo $NETCDF_PATH
```
Comment (aburrell) [inline .github/workflows/main.yml:69]:
```suggestion
```

**Pull request merged**